### PR TITLE
getLayers(type)

### PR DESCRIPTION
**Pull request by vcoindet**
Return the layers in the scene.
For now, type can be 'colorLayers' or 'elevationLayers', if not specified : returns all the layers.

### CODE REVIEW

**Comment by vcoindet** [comment]
Rebased and Ready

File: src/Core/Commander/Interfaces/ApiInterface/ApiGlobe.js [hunk 1]
```diff
@@ -917,6 +917,25 @@ ApiGlobe.prototype.removeEventListenerLayerChanged = function removeEventListene
...
 
+/**
+ * Return all the layer in the scene.
+ * The type can be 'colorLayers' or 'elevationLayers'. If the type is not specified, the function return all the layers.
+ * @param {type} Type - The type of the layers wanted.
+ */
+
+ApiGlobe.prototype.getLayers = function getLayers(type) {
+    const lc = this.scene.getMap().layersConfiguration;
+    const layers = [];
```
Comment (gchoqueux) [inline src/Core/Commander/Interfaces/ApiInterface/ApiGlobe.js]: remove `layers`,  it's useless to reconstruct a new Array

**Review by gchoqueux — CHANGES_REQUESTED** [review]
do in this PR :

ApiGlobe.prototype.getLayerById()

**Comment by vcoindet** [comment]
Changes done @gchoqueux 
rebased and ready

**Comment by vcoindet** [comment]
rebased

**Comment by vcoindet** [comment]
rebased

**Comment by vcoindet** [comment]
rebased and ready @gchoqueux

File: src/Core/Commander/Interfaces/ApiInterface/ApiGlobe.js [hunk 2]
```diff
@@ -915,6 +915,38 @@ ApiGlobe.prototype.removeEventListenerLayerChanged = function removeEventListene
...
+/**
+ * Return all the layer in the scene.
+ * The type can be 'colorLayers' or 'elevationLayers'. If the type is not specified, the function return all the layers.
+ * @param {type} Type - The type of the layers wanted.
+ */
+
+ApiGlobe.prototype.getLayers = function getLayers(type) {
+    const lc = this.scene.getMap().layersConfiguration;
+    if (type === undefined) {
+        return [...lc.getColorLayers(), ...lc.getElevationLayers()];
```
Comment (gchoqueux) [inline src/Core/Commander/Interfaces/ApiInterface/ApiGlobe.js]: an geometry layer?

File: src/Core/Commander/Interfaces/ApiInterface/ApiGlobe.js [hunk 3]
```diff
@@ -915,6 +915,38 @@ ApiGlobe.prototype.removeEventListenerLayerChanged = function removeEventListene
...
+};
+
+/**
+ * Return a layer by its id.
+ * @param {id} ID - The id of the layer wanted.
+ */
+
+ApiGlobe.prototype.getLayerById = function getLayerById(pId) {
+    const lc = this.scene.getMap().layersConfiguration;
+    const tabLayers = [...lc.getColorLayers(), ...lc.getElevationLayers()];
```
Comment (gchoqueux) [inline src/Core/Commander/Interfaces/ApiInterface/ApiGlobe.js]: * factorize the aggregate in  `layersConfiguration`
* geometry layer?

File: src/Core/Commander/Interfaces/ApiInterface/ApiGlobe.js [hunk 4]
```diff
@@ -915,6 +915,38 @@ ApiGlobe.prototype.removeEventListenerLayerChanged = function removeEventListene
...
+    for (const i of tabLayers) {
```
Comment (gchoqueux) [inline src/Core/Commander/Interfaces/ApiInterface/ApiGlobe.js]: use `Array.find()`

**Review by gchoqueux — CHANGES_REQUESTED** [review]

**Comment by vcoindet** [comment]
rebased and ready @gchoqueux

File: src/Core/Commander/Interfaces/ApiInterface/ApiGlobe.js [hunk 5]
```diff
@@ -915,6 +915,36 @@ ApiGlobe.prototype.removeEventListenerLayerChanged = function removeEventListene
...
+ * Return all the layer in the scene.
+ * The type can be 'colorLayers', 'elevationLayers' and 'geometryLayers'. If the type is not specified, the function return all the layers.
+ * @param {type} Type - The type of the layers wanted.
+ */
+
+ApiGlobe.prototype.getLayers = function getLayers(type) {
+    const lc = this.scene.getMap().layersConfiguration;
+    if (!type) {
+        return lc.getLayers();
+    } else if (type === 'colorLayers') {
```
Comment (gchoqueux) [inline src/Core/Commander/Interfaces/ApiInterface/ApiGlobe.js]: just replace by color, elevation and geometry

Comment (vcoindet) [reply src/Core/Commander/Interfaces/ApiInterface/ApiGlobe.js]: done

**Comment by vcoindet** [comment]
rebased and ready

**Review by gchoqueux — APPROVED** [review]

**Pull request merged**